src/components/Wallet/WalletDeposit.jsx

```diff
@@ -50,47 +53,41 @@ class WalletDeposit extends React.Component {
     clearInterval(depositTimer);
   }
 
-  getDepositableNodes = () => {
-    const { depositNodes } = this.props;
-    const nodes = Object.values(depositNodes.nodes);
-    const depositable = [];
-
-    for (let i = 0; i < nodes.length; i += 1) {
-      const node = nodes[i];
-      if (node.balanceSats.isEqualTo(0) && !node.addressUsed) {
-        depositable.push(node);
-      }
-    }
-    return depositable;
-  };
-
   getNextDepositAddress = () => {
     const { depositIndex } = this.state;
     this.setState({ depositIndex: depositIndex + 1 });
     setTimeout(this.getDepositAddress, 0);
   };
 
   getDepositAddress = () => {
-    const { network, client, updateDepositSlice } = this.props;
+    const {
+      network,
+      client,
+      updateDepositSlice,
+      depositableSlices,
+    } = this.props;
     const { depositIndex } = this.state;
-    const depositableNodes = this.getDepositableNodes();
-    if (depositIndex < depositableNodes.length)
+
+    if (depositIndex < depositableSlices.length)
```

Comment:
This is coming from selector (and therefore memoized) now instead of calculated on demand.